libs/platform/src/lib/components/form/step-input/base.step-input.ts

```diff
@@ -0,0 +1,477 @@
+        public source: T,
+        /** The new value of a control. */
+        public payload: K
+    ) {}
+}
+
+export const enum StepInputAlign {
+    Left = 'left',
+    Center = 'center',
+    Right = 'right'
+}
+
+export type StepInputStepFunctionAction = 'increase' | 'decrease';
+export type StepInputStepFunction = (value: number, action: StepInputStepFunctionAction) => number;
+
+const ALIGN_INPUT_OPTIONS_LIST = [StepInputAlign.Left, StepInputAlign.Center, StepInputAlign.Right];
+
+/**
+ * StepInputComponent is a base abstract class that should be used
+ * to create type specific StepInput components such as number, money, unitOfMeasure
+ * This holds base Step Input functionality that can be abstracted
+ */
+@Directive()
+export abstract class StepInputComponent extends BaseInput implements OnInit {
+    /** content Density of element. cozy | compact */
+    @Input()
+    set contentDensity(contentDensity: ContentDensity) {
+        this._contentDensity = contentDensity;
+        this.isCompact = this._contentDensity === 'compact';
+    }
+
+    /** Sets input value */
+    @Input()
+    get value(): number {
+        return super.getValue();
+    }
+    set value(value: number) {
+        if (value !== this._value) {
+            super.setValue(value);
+            this._calculateCanDecrementIncrement();
+        }
+    }
+
+    /** Sets minimum value boundary */
+    @Input()
+    set min(min: number) {
+        this._min = !isNaN(min) ? min : -Number.MAX_VALUE;
+        this._calculateCanDecrement();
+    }
+    get min(): number {
+        return this._min;
+    }
+
+    /** Sets maximum value boundary */
+    @Input()
+    set max(max: number) {
+        this._max = !isNaN(max) ? max : Number.MAX_VALUE;
+        this._calculateCanIncrement();
+    }
+    get max(): number {
+        return this._max;
+    }
+
+    /** Sets input step value */
+    @Input()
+    set step(step: number) {
+        this._checkAndThrowErrorIfStepDoesntMatchPrecision(this.precision, step);
+        this._step = step;
+        this._calculateCanDecrementIncrement();
+    }
+    get step(): number {
+        return this._step;
+    }
+
+    /** Custom function to calculate step dynamically */
+    @Input()
+    set stepFn(stepFn: StepInputStepFunction) {
+        this._stepFn = stepFn;
+        this._calculateCanDecrementIncrement();
+    }
+
+    /** Number of digits after the decimal point */
+    @Input()
+    set precision(precision: number) {
+        this._checkAndThrowErrorIfStepDoesntMatchPrecision(precision, this.step);
+        this._precision = precision;
+    }
+    get precision(): number {
+        return this._precision;
+    }
+
+    /** Sets largeStep multiplier */
+    @Input()
+    largerStep = 2;
+
+    /** Hides -/+ icons and shows labels */
+    @Input()
+    showLabels = false;
+
+    /** Horizontally aligns value inside input */
+    @Input()
+    set align(align: StepInputAlign) {
+        this._align$.next(align);
```

Comment:
The same here with the stateChanges and other more significant inputs that can change after component is initialized.

Reply:
Why do we need to notify parent when `align` changes?
`Align` does not influence on form state, it just changes style in the template

Reply:
We need to inform a parent to trigger change detector if some state change after component is created. You are right that align is going to be set only once we are not going to change it afterwards